### ISSUE TITLE
[Bug] [Android] ClearButton initially not showing up

### ISSUE THREAD

**Issue by LarsBehl** [dotnet/maui#3379]
### Description
`ClearButton` won't be displayed on `Entry` while editing if `Entry.Text` is initially empty, when setting `ClearButtonVisibility` to `WhileEditing`.

#### Note
If the focus to the `Entry` is lost and it gets refocused, the `ClearButton` shows up (if the `Entry` contains text) but has no effect.

### Steps to Reproduce

1. Add `Entry` to some view
2. Set `ClearButtonVisibility` to `WhileEditing`
3. Type in some text

### Expected Behavior
The `ClearButton` shows up as soon as the first character is entered into the `Entry`

### Actual Behavior
The `ClearButton` does not show up

### Basic Information

- Version with issue: 6.0.101-preview.10.2068
- Last known good version: -
- IDE: Visual Studio 2022 Version 17.1.0 Preview 1.0
- Platform Target Frameworks: 
  - Android: 28

#### Expected
![image](https://user-images.githubusercontent.com/26459044/141646237-d875fa39-c66b-4539-9ccf-82bc8d8c6a91.png)

#### Actual
![image](https://user-images.githubusercontent.com/26459044/141646222-bcf9b643-0d37-484f-916b-0cdc96d55633.png)

### Reproduction Link
The issue is present [here](https://github.com/LarsBehl/StatusApp/blob/0713a6f7a38f6dac382f8c644d07042317e8dcf4/Views/SettingsView.xaml)

**Comment by jsuarezruiz** [comment]
Cannot reproduce the issue using the main branch (upcoming Preview):
![image](https://user-images.githubusercontent.com/6755973/141735922-d2637f85-5b12-43c2-841b-5e34f77e66ac.png)

I am going to compare the Entry Handler code between Preview 10 and the current version.

**Comment by LarsBehl** [comment]
I just installed the workload of the main branch and could still reproduce the issue:

![image](https://user-images.githubusercontent.com/26459044/141817343-ab6e7282-0bf8-429f-9a6b-a7e84b433cb7.png)

The version that installed with the workload is 6.0.101-preview.11.2167.
The device is a Pixel 3a.

I've just created a new maui application and replaced the content of the `MainPage` with a `VerticalStackLayout` containing two `Entry`s. For the first one I've set `ClearButtonVisibility` to `WhileEditing` in the xaml file.
